TrueCraft.Core.Test/World/SugarCaneDecoratorTests.cs

```diff
@@ -0,0 +1,461 @@
+    {
+        public override double Value2D(double x, double y)
+        {
+            return x > 5 ? 0 : (y > 5 ? 0 : 1);
+        }
+
+        public override double Value3D(double x, double y, double z)
+        {
+            double value2d = Value2D(x, y);
+
+            return 1;
+        }
+    }
+
+    public class PrimeSugarCaneGrowingSeasonChunk : IChunk
+    {
+        public void Dispose()
+        {
+            throw new NotImplementedException();
+        }
+
+        public event EventHandler Disposed;
+        public int X => 6;
+        public int Z => 6;
+        public int MaxHeight => 7;
+        public Coordinates2D Coordinates { get; set; }
+        public bool IsModified { get; set; }
+        public bool LightPopulated { get; set; }
+        public int[] HeightMap => Enumerable.Repeat(1, Chunk.Width * Chunk.Height).ToArray<int>();
+        public byte[] Biomes => Enumerable.Repeat(new SwamplandBiome().ID, Chunk.Width*Chunk.Height).ToArray<byte>();
+
+        public DateTime LastAccessed { get; set; }
+        public byte[] Data { get; }
+        public bool TerrainPopulated { get; set; }
+        public Dictionary<Coordinates3D, NbtCompound> TileEntities { get; set; }
+        public NibbleSlice Metadata { get; }
+        public NibbleSlice BlockLight { get; }
+        public NibbleSlice SkyLight { get; }
+        public IRegion ParentRegion { get; set; }
+        public int GetHeight(byte x, byte z)
+        {
+            // Pretty sure this is always one since we have a flat surface 
+            // (Unless water heights are reported differently)
+            return 1;
+        }
+
+        public void UpdateHeightMap()
+        {
+            throw new NotImplementedException();
+        }
+
+
+        private static byte MapXZToTestBlock(int X, int Z)
+        {
+            // Sand on the outer left hand side
+            if (X == 0)
+            {
+                return GrassBlock.BlockID;
+            }
+            
+            // Sand on the outer right hand side
+            if (X == 5)
+            {
+                return SandBlock.BlockID;
+            }
+
+            if (Z == 0)
+                return WaterBlock.BlockID;
+
+            if (Z == 3)
+                return StationaryWaterBlock.BlockID;
+
+            switch (X)
+            {
+                case 1:
+                    return GrassBlock.BlockID;
+                case 2:
+                    return DirtBlock.BlockID;
+                case 3:
+                    return SandBlock.BlockID;
+                case 4:
+                    return StoneBlock.BlockID;
+            }
+
+            // What's in that block? 
+            // Back at me. 
+            // I have it, it's a block that doesn't matter because it's beyond our test grid.  
+            // Look again
+            // The block is now diamonds. 
+            // Anything is possible when you're in a test mock
+            // I'm on a pig. 
```

Comment:
Can we take this a little bit more seriously? This should only include as much code as is necessary to test your changes and nothing else.

Reply:
Will fix.